tests/python/test_gui.py

```diff
@@ -0,0 +1,30 @@
+from taichi import make_temp_file
+import taichi as ti
+import numpy as np
+import pytest
+
+
+@ti.host_arch_only
+@pytest.mark.parametrize('dtype', [ti.u8, ti.f32])
+def test_save_image_without_window(dtype):
+    n = 255
+    pixels = ti.field(dtype=dtype, shape=(n, n, 3))
+
+    @ti.kernel
+    def paint(c: dtype):
+        for i, j, k in pixels:
+            pixels[i, j, k] = c
+
+    gui = ti.GUI("Test", res=(n, n), show_gui=False)
+    for i in [0, 32, 64, 128, 255]:
+        if dtype is ti.u8:
+            paint(i)
+        else:
+            paint(i * 1.0 / n)
+        gui.set_image(pixels)
+        image_path = make_temp_file(suffix='.jpg')
```

Comment:
```suggestion
        image_path = make_temp_file(suffix='.png')
```

Let's use a lossless format here. (I know jpg images with a single color, as a special case, is probably lossless, but let's try not to confuse people with special cases :-)